pkg/core/index_document.go

```diff
@@ -250,3 +291,28 @@ func (index *Index) FindID(id string) (*bluge.Writer, error) {
 	}
 	return nil, errors.ErrorIDNotFound
 }
+
+// isDateProperty returns true if the given value matches the default date format.
+func isDateProperty(value string) (string, bool) {
+	layout := detectTimeLayout(value)
+	_, err := time.Parse(layout, value)
+
+	return layout, err == nil
+}
+
+// detectTimeLayout tries to figure out the correct layout of the input date.
+func detectTimeLayout(value string) string {
+	layout := ""
+	switch {
+	case len(value) == 19 && strings.Index(value, " ") == 10:
+		layout = "2006-01-02 15:04:05"
+	case len(value) == 19 && strings.Index(value, "T") == 10:
+		layout = "2006-01-02T15:04:05"
+	case len(value) == 25 && strings.Index(value, "T") == 10:
+		layout = time.RFC3339
+	case len(value) == 29 && strings.Index(value, "T") == 10 && strings.Index(value, ".") == 19:
+		layout = "2006-01-02T15:04:05.999Z07:00"
+	}
```

Comment:
I think here need more check, we can not simplely use `len(value) == 19 && strings.Index(value, " ") == 10` then let `format="2006-01-02 15:04:05"`, like add more check about `-` and `:`

Reply:
`len(value) == 19 && strings.Index(value, " ")`
maybe it is `2006/01/02 15:04:05`, or `Jun 1 2002 15:04:05` not only `2006-01-02 15:04:05`

Reply:
The problem with `2006/01/02 15:04:05` is that some people may write the month in the third number place holder whereas others write it in the second number place holder.

And `Jun 1 2002 15:04:05` seems to be a real edge case – if the user ever wanted to use such a date format, automatic detection of such a format may not be required.

What type of formats do you think should be supported additionally?

Reply:
It's okay, just keep this, if user report need other formats we will update it.